grails-app/domain/org/pih/warehouse/product/Product.groovy

```diff
@@ -678,22 +680,44 @@ class Product implements Comparable, Serializable {
     }
 
     List getUoms() {
-       return packages.collect { [uom: it.uom.code, quantity: it.quantity] }.unique()
+        return packages.collect { [uom: it.uom.code, quantity: it.quantity] }.unique()
     }
 
     String getTranslatedNameWithLocaleCode() {
         String localeTag = LocalizationUtil.localizationService.getCurrentLocale().toLanguageTag().toUpperCase()
         return "${name}${translatedName ? " (${localeTag}: ${translatedName})" : ''}"
     }
 
-    String getTranslatedName() {
-        Locale currentLocale = LocalizationUtil.localizationService.getCurrentLocale()
+    Synonym getSynonymByLocale(SynonymTypeCode synonymTypeCode, Locale locale) {
```

Comment:
If nothings has changed I believe that only **DISPLAY_NAME** has a validation for having one **DISPLAY_NAME** per locale and other typeCodes can have multiple.

So if we are making this `getSynonymByLocale()` function more generic, that it can return synonyms of other SynonymTypeCodes and not just **DISPLAY_NAME**, it should probably return a `List<Synonym>` because there might be cases where a product has multiple `ALTERNATIVE_NAME` or `BRAND_NAME` in one locale.

Reply:
Good point. Thanks for bringing that up. I'll try to figure out how to handle that.

Reply:
This would actually be a great place for some unit tests.